src/components/Map/PlantingSiteMap.tsx

```diff
@@ -104,15 +118,27 @@ export default function PlantingSiteMap(props: PlantingSiteMapProps): JSX.Elemen
   );
 
   // fetch polygons and boundaries
+  const [includedLayers, setIncludedLayers] = useState<MapLayer[] | undefined>(layerOptions);
   useEffect(() => {
     const fetchPlantingSite = () => {
       const site = extractPlantingSite(plantingSite);
       const zones = extractPlantingZones(plantingSite);
       const subzones = extractSubzones(plantingSite);
 
+      const includedSources = new Array<MapSource>();
+      if (includedLayers?.includes('Sub-Zones')) {
+        includedSources.push(subzones);
+      }
+      if (includedLayers?.includes('Zones')) {
+        includedSources.push(zones);
+      }
+      if (includedLayers?.includes('Planting Site')) {
+        includedSources.push(site);
+      }
+
       const newMapOptions = {
         bbox: MapService.getPlantingSiteBoundingBox(plantingSite),
-        sources: [site, subzones, zones],
+        sources: includedSources,
```

Comment:
This is great and works! 
Food for thought if it helps - extend the entity/source props with a visible/boolean flag  - which can then be filtered in the map. This is one way I have seen/done layer based viz in the past.
We could flip on/off specific layers... or on/off specific data.
On one hand it is repopulating data sources, on another it is configuring view options of data sources. Pros/cons to both. Whichever you think scales better for the additional functionalities like search results based viewing etc.
What we have here is simple/correct and works - just wanted to share data/view separation idea if it clicks.

Reply:
The layer versus search are actually orthogonal.. I guess search will always be changes to the data source. My bad.

Reply:
I actually like the visible/non-visible switch option. I'll see about implementing that

Reply:
Hmm, not immediately getting this to work. I'll revisit it later

Reply:
No worries, thanks for giving this a try.